pkg/config/template.go

```diff
@@ -47,3 +60,60 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMController holds the details needed for the configmap controller
+type TemplateCMController struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates map[string]*templateapi.Template
+}
+
+// NewTemplateCMController creates a new TemplateCMController
+func NewTemplateCMController(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates map[string]*templateapi.Template) *TemplateCMController {
+	return &TemplateCMController{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMController) CreateCMTemplates() []error {
+	var errors []error
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+
+	for filename, template := range c.templates {
+		buf := new(bytes.Buffer)
+
+		// Encode the object to YAML.
+		err := s.Encode(template, buf)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cm := &v1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("rehearse-%d-%s", c.prNumber, template.Name)},
```

Comment:
Can we also label the CMs, just like we label the rehearsal jobs.

Reply:
that could be useful when we delete all the configmaps. If we delete them by label, we won't have any forgotten cm